src/main/java/org/mockito/listeners/VerificationListener.java

```diff
@@ -0,0 +1,22 @@
+package org.mockito.listeners;
+
+import org.mockito.internal.verification.api.VerificationData;
+import org.mockito.verification.VerificationMode;
+import org.mockito.verification.VerificationSucceededEvent;
+
+/**
+ * This listener can be notified of verify invocations on a mock.
+ *
+ * For this to happen, it must be registered using {@link org.mockito.internal.progress.MockingProgress#addListener(MockitoListener)}.
+ */
+public interface VerificationListener extends MockitoListener {
+    /**
+     * Called when a verification completes successfully without throwing an exception.
+     *
+     * @param verificationSucceededEvent contains all information about the event.
+     *                                   Consisting of the mock that was verified on,
+     *                                   the {@link org.mockito.verification.VerificationMode} used
+     *                                   and the {@link org.mockito.internal.verification.api.VerificationData}
+     */
+    void onVerificationSucceeded(VerificationSucceededEvent verificationSucceededEvent);
```

Comment:
1. More generic API that also covers verification failure would be better. 
2. 'onMockVerified' is more consistent in naming with existing 'onMockCreated'
3. Why don't we keep the API simple and ditch the 'event' object.

For example:

```java
//failure is nullable
onMockVerified(Object mock, VerificationMode mode, VerificationData data, Throwable failure)
```